src/app/d2-vendors/VendorItemComponent.tsx

```diff
@@ -106,19 +108,24 @@ export default class VendorItemComponent extends React.Component<Props, {}> {
     } else {
 
       let reviewData: DimWorkingUserReview | null = null;
+      let dimItem: DimItem | null = null;
 
       if (trackerService) {
         reviewData = trackerService.getD2ReviewDataCache().getRatingData(undefined, item.itemHash);
 
         if (reviewData && !reviewData.reviews) {
-          const reviewsData = await trackerService.getItemReviewAsync(item.itemHash);
-
-          Object.assign(reviewData, reviewsData);
+          trackerService.getItemReviewAsync(item.itemHash).then((reviewsData) => {
```

Comment:
As a takeaway for me, the trackerService should provide a simpler affordance to fetch rating data. This ugly service->cache->fetch flow is from an earlier iteration.

I tried to move this all into the service, but the data flows are kind of... entangled.